### PR TITLE
PC-11626-rspec-product-controller-tests

### PR DESCRIPTION
**Pull request by aminasaif0**

### CODE REVIEW

File: spec/controller/product_spec.rb [hunk 1]
```diff
@@ -18,4 +23,79 @@
...
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid attributes' do
+      it 'creates a new product' do
+        expect {
+          post :create, params: { product: FactoryBot.attributes_for(:product, category_id: category.id) }
+        }.to change(Product, :count).by(1)
+      end
```
Comment (saadali7865) [inline spec/controller/product_spec.rb]: Test the created object as well. 
Like 
Expected Product name
Expected Product Category

File: spec/controller/product_spec.rb [hunk 2]
```diff
@@ -18,4 +23,79 @@
...
+        expect(response).to render_template(:edit)
+        expect(assigns(:product).errors).not_to be_empty
+      end
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    it 'destroys the product' do
+      product
+      sleep 1
```
Comment (saadali7865) [inline spec/controller/product_spec.rb]: Why sleep 1 ?

File: spec/controller/product_spec.rb [hunk 3]
```diff
@@ -18,4 +23,79 @@
...
+      expect{
+        delete :destroy, params: {id: product.id}
+      }.to change(Product,:count).by(-1)
```
Comment (saadali7865) [inline spec/controller/product_spec.rb]: Instead of just checking the count. Check if product with that Id exist or not

**Review by saadali7865 — CHANGES_REQUESTED** [review]

File: spec/controller/product_spec.rb (hunk 2 above)
Comment (aminasaif0) [reply spec/controller/product_spec.rb]: due to issue with db refresh, I found 2 solutions for delay 
1. use sleep 1 
3. Product.__elasticsearch__.refresh_index! do this

**Review by saadali7865 — APPROVED** [review]

**Pull request merged**